### PR TITLE
Added plugin for Newport XPS Q8

### PR DESCRIPTION
**Pull request by sq-lumin**
Added plugin for the Newport XPS-Q8 Motion Controller. 
Tested on Windows 11 with PyMoDAQ 4.1.0 with a single translation stage.
Only stop_motion is not implemented as I haven't found a corresponding method in the wrapper given by Newport.
Thank you for the review & merge

### CODE REVIEW

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    DataActuator  # common set of parameters for all actuators
+from pymodaq.utils.daq_utils import ThreadCommand # object used to send info back to the main thread
+from pymodaq.utils.parameter import Parameter
+from qtpy.QtCore import QThread
+from pymodaq_plugins_newport.hardware import XPS_Q8_drivers 
+import sys
+
+from time import perf_counter_ns
+
+class XPSPythonWrapper():
```
Comment (seb5g) [inline src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py:11]: put this wrapper in the hardware folder which is dedicated for that and keep here only the plugin features

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        
+        #Definition of the stage
+        self._group = group 
+        self._positioner = positionner
+        self._full_positionner_name = f'{group}.{positionner}'
+        
+        #Some required initialisation steps
+        self._initCommands()
+            
+    def _initCommands(self):
```
Comment (seb5g) [inline src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py:34]: try to use only lowercase in methods using underscore to separate words

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+from pymodaq_plugins_newport.hardware import XPS_Q8_drivers 
+import sys
+
+from time import perf_counter_ns
+
+class XPSPythonWrapper():
+    """Simplified XPS wrapper, calls methods from the wrapper given by Newport. See XPS_Q8_drivers"""
+    
+    def __init__(self, ip:str = None, port:int = None, group:str = None, positionner:str = None, plugin = None):
+        #init the wrapper given by Newport and some attributes
```
Comment (seb5g) [inline src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py:15]: please add docstring precising what should be your paramaters

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+        self.myxps = XPS_Q8_drivers.XPS()   #Instanciate the driver from Newport
+        
+        #keep a ref of the plugin to emit (error) messages
+        self._plugin = plugin
```
Comment (seb5g) [inline src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py:19]: your driver should ingeneral be unaware of your plugin as in a nice world some other people could use it for another task in another program. if you want to pass errors, use logger:
```
from pymodaq.utils.logger import set_logger, get_module_name
logger = set_logger(get_module_name(__file__))

logger.error('my error')
```
or you can raise an error and catch it in your plugin to transform it into a emit_status

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+            if (errorCode != 0):
+                self.displayErrorAndClose(errorCode, 'GroupKill')
+
+            #Initialize
+            [errorCode, returnString] = self.myxps.GroupInitialize(self.socketId, self._group) 
+            if (errorCode != 0):
+                self.displayErrorAndClose(errorCode, 'GroupInitialize')
+            
+            #Home search
+            self.moveHome()
```
Comment (seb5g) [inline src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py:55]: use only lowercase, please apply this everywhere

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+        Returns
+        -------
+        info: str
+        initialized: bool
+            False if initialization failed otherwise True
+        """
+
+        self.controller = self.ini_stage_init(old_controller=controller,
+                                              new_controller=XPSPythonWrapper(
+                                                  ip = self.settings.child('xps_ip_address').value(),
```
Comment (seb5g) [inline src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py:237]: to access only value of a parameter use the shortcut:
```
self.settings['xps_ip_address']
```

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+        """Sets a new port. Does not automatically try to connect with it, call retryConnection."""
+        self._port = port
+        
+    def retryConnection(self):
+        """Closes the connection and runs the init sequence again.
+        Called by the plugin after any change to the IP address or port parameters."""
+        self.closeTCPIP()
+        self._initCommands()
+    
+class DAQ_Move_Newport_XPS_Q8(DAQ_Move_base):
```
Comment (seb5g) [inline src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py:146]: For the class name I would rather use:
 DAQ_Move_NewportXpsQ8

or directly
DAQ_Move_XpsQ8

**Review by seb5g — CHANGES_REQUESTED** [review]
thanks for the PR and sorry for the delay
please apply my comments in the review

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py (hunk 1 above)
Comment (sq-lumin) [reply src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py]: [476e677](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/476e677c9edae544c4f6c77cb10a9a425fe91602) moved to new file xps_q8_simplified.py so as to not modify the driver provided by the constructor (XPS_Q8_drivers.py)

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py (hunk 3 above)
Comment (sq-lumin) [reply src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py]: done on commit [32d9cd8](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/32d9cd836e5e27d17f38f65f06bc1b10124783bd)

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py (hunk 4 above)
Comment (sq-lumin) [reply src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py]: done on commit [abcc736](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/abcc73647d8281f5c30ddc2ed09147e4a6cc00bd). I chose to raise a custom "XPSError" to keep the hardware wrapper independant from pymodaq, and to catch those in the plugin. Thanks for the suggestion

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py (hunk 6 above)
Comment (sq-lumin) [reply src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py]: thanks, done on commit [6390e14](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/6390e14ec43556b665ca15bfbb67280992660fe8)

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py (hunk 7 above)
Comment (sq-lumin) [reply src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py]: Changed to DAQ_Move_XpsQ8 on commit [8cd0fe3](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/8cd0fe3b456c37e5a4d1f27d5da8fdf59554b56d)

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py (hunk 5 above)
Comment (sq-lumin) [reply src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py]: done mainly on commits [e909d03](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/e909d033ddb3efb4524667b5bc990bc3cf0de203) & [f0e37fd](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/f0e37fde78a425566ce42e16cc4371c136766337)

**Comment by sq-lumin** [comment]
Thank you for the review and sorry for the delay as well. I applied the requested changes, and tested that the plugin still works on an XPS Q8 with pymodaq 4.2.4 on windows 11.
Please let me know if you have any additionnal requests

File: src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py (hunk 1 above)
Comment (sq-lumin) [reply src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py]: Sorry for the messy commit, most of the changes in the daq_move file are from my linter

**Review by seb5g — APPROVED** [review]

**Comment by seb5g** [comment]
thanks for the PR, I will publish a release with your code available for all!

**Pull request merged**